File: Java/src/1431_KidsWiththeGreatestNumberofCandies/KidsWiththeGreatestNumberofCandies.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int max = candies[0];
        for(int i = 1 ; i< candies.length ; i++){
            if(max < candies[i]) max = candies[i];
        }
        ArrayList<Boolean> ans = new ArrayList<>();
        for(int i = 0 ; i < candies.length ; i++){
            if(max <= candies[i]+extraCandies)
                ans.add(true);
            else
                ans.add(false);
        }
        return ans;
    }
}

File: golang/1431_KidsWiththeGreatestNumberofCandies/kidsWiththeGreatestNumberofCandies.go
func kidsWithCandies(candies []int, extraCandies int) []bool {
	max := candies[0]
	for i := range candies {
		if max < candies[i] {
			max = candies[i]
		}
	}
	ans := []bool{}
	for i := range candies {
		if candies[i]+extraCandies >= max {
			ans = append(ans, true)
		} else {
			ans = append(ans, false)
		}
	}
	return ans
}